dietpi/dietpi-software

```diff
@@ -4963,23 +4962,32 @@ _EOF_
 			echo -e 'Package: *\nPin: release n=sid\nPin-Priority: 99' > /etc/apt/preferences.d/dietpi-wireguard
 			G_AGUP
 
-			# iptables required to forward incoming VPN traffic to internet interface
-			# qrencode required to create client config QR code on console
-			G_AGI $kernel_packages wireguard iptables qrencode
+			G_AGI $kernel_packages
+			# Prevent kernel updates from breaking compiled module.
+			G_RUN_CMD apt-mark hold $kernel_packages
+
+			# Kernel mismatch, reboot required before build.
+			if [[ $kernel_version_active != $(ls /lib/modules/) ]]; then
 
-			# This can only match on RPi, if "rpi-update" was detected
-			# - If APT updates did not override "rpi-update" kernel version, headers might mismatch and WireGuard module missing
-			if [[ $old_firmware == $(ls /lib/modules/) ]]; then
+				G_WHIP_MSG 'Wireguard:\n\nKernel update detected, a reboot is required before Wireguard module can be built with latest kernel.\n\nOnce the system has rebooted, reselect Wireguard for installation to continue.'
+				aSOFTWARE_INSTALL_STATE[$software_id]=0
```

Comment:
@Fourdee 
Reboot not required here. The module will be build for all installed kernels, regardless of which one is active. So after reboot (which is done anyway after dietpi-software), WireGuard will work.

Reply:
@MichaIng 

> Reboot not required here. The module will be build for all installed kernels, regardless of which one is active. So after reboot (which is done anyway after dietpi-software), WireGuard will work.

Excellent, i'll make the change 👍

Reply:
@MichaIng 

> WireGuard rebuilds the module automatically, when the kernel packages are updated due to some APT trigger, or did this not work on RPi (was the case on x86_64 at least)? 

Will need to verify this is case.